app/assets/javascripts/layout_ace_editor.js

```diff
@@ -0,0 +1,117 @@
+
+  /**
+   * Builds the editor within the given container.
+   *
+   * @param {jQuery} $container The container to build the editor in.
+   * @param {Object} options The options for the editor.
+   * @returns {Object} The Ace editor instance.
+   */
+  function build_editor($container, options) {
+    var editor = ace.edit($container[0]);
+    editor.setTheme('ace/theme/' + options['theme']);
+    editor.getSession().setMode('ace/mode/' + options['lang']);
+
+    return editor;
+  }
+
+  /**
+   * Finds or builds an editor container from the given jQuery element.
+   *
+   * @param {jQuery} $element The element to find or build an Ace editor for.
+   * @param {Object} options The options to use when building the element.
+   * @returns {jQuery}
+   */
+  function find_or_build_editor_container($element, options) {
+    var ace_data = $element.data('ace');
+    if (ace_data) {
+      return ace_data['container'];
+    }
+
+    var $container = build_editor_container($element[0].id);
+    $container.insertAfter($element);
+    $container.height($element.height());
+
+    var editor = build_editor($container, options);
+    editor.on('change', function() {
+      $element.val(editor.session.getValue());
+    });
+    editor.session.setValue($element.val());
+    assign_editor($element, $container, editor);
+
+    return $container;
+  }
+
+  /**
+   * Assigns the Ace configuration to the given element.
+   *
+   * @param {jQuery} $element The element which has the Ace editor associated with.
+   * @param {jQuery} $container The container element for the Ace editor.
+   * @param {Object} ace The Ace editor created from `ace.edit`.
+   */
+  function assign_editor($element, $container, ace) {
+    $element.data('ace', build_ace_options($container, ace));
+  }
+
+  /**
+   * Builds the Ace configuration object.
+   *
+   * @param {jQuery} $container The container for the Ace editor.
+   * @param {Object} ace The Ace editor created.
+   * @returns {Object}
+   */
+  function build_ace_options($container, ace) {
+    return {
+      'container': $container,
+      'editor': ace
+    };
+  }
+
+  /**
+   * Redirects all labels from the old textarea to point to the new editor.
+   * @param {jQuery} $from The textarea to redirect the labels from.
+   * @param {jQuery} $editor The container for the Ace editor to redirect the labels to.
+   */
+  function reassign_labels_to_ace($from, $editor) {
+    var from_id = $from[0].id;
+    var $editor_textarea = $('textarea.ace_text-input', $editor);
+    var editor_textarea_id = $editor_textarea[0].id = 'ace_textarea_' + from_id;
+    $('label[for="' + from_id + '"]').attr('for', editor_textarea_id);
+  }
+
+  $.fn.ace = function(options) {
+    options = $.extend({}, $.fn.ace.defaults, options);
+
+    return this.each(function() {
+      var $this = $(this);
+      var element_options = $.extend({}, options, { 'lang': $this[0].lang });
+      var $editor = find_or_build_editor_container($this, element_options);
```

Comment:
Identifier 'find_or_build_editor_container' is not in camel case.<br>Identifier 'element_options' is not in camel case.